Azure/azure-cli
# [CDN] Add private link fields to origin

**Description**  

This PR adds support for the private link feature in Azure CDN. It modifies commands `az cdn endpoint create` to add private link fields, adds a new command `az cdn origin update` to allow modifying private link fields of the origin on an existing endpoint, and bumps the SDK version to use the new CDN API version 2020-04-15 which supports private link fields.

Please note that a few lines of an existing test, `CdnWafEndpointLinkScenarioTest`, are modified/commented out to work around a bug in our service - this bug only impacts a preview feature, and we have already implemented a fix which will be deployed soon. If this PR is merged before the bugfix is deployed, we will promptly submit a new PR to reenable the test.

**Testing Guide**  

Create an endpoint with a private origin:
```
az cdn endpoint create -g group -n endpoint --profile-name profile --origin www.example.com 80 443 /subscriptions/subid/resourcegroups/rg1/providers/Microsoft.Network/privateLinkServices/pls1 eastus 'Please approve this request'
```

Make an existing endpoint's origin private:
```
az cdn origin update --g group --profile-name profile --endpoint-name endpoint -n origin --http-port 80 --https-port 443 --private-link-resource-id  --private-link-location EastUS --private-link-approval-message 'Please approve this request'
```
---

This checklist is used to make sure that common guidelines for a pull request are followed.
